### ISSUE TITLE
Cannot connect to AMT Console

### ISSUE THREAD

**Issue by ib-mlatin** [Ylianst/MeshCentral#2703]
I just upgraded to MeshCentral v0.8.48 and it seems like that broke various functions in MeshCentral. Thankfully it still works through MeshCommander. I am using a reverse proxy using Apache2.

The first issue I am having is nothing happens when clicking on any device, whether it be a CIRA/AMT device or an agent device. The bottom error repeats as you click more devices. The top two errors show when loading any page. The device does load when you click on it twice for whatever reason. 

```
Uncaught TypeError: Cannot read property 'id' of undefined
```

![image](https://user-images.githubusercontent.com/60354324/119890787-dedd9900-befd-11eb-8bc6-a091fffe0a0a.png)

[meshcentral-device-1622146278358.log](https://github.com/Ylianst/MeshCentral/files/6556535/meshcentral-device-1622146278358.log)

When clicking on the connect button, the below errors occur. Nothing happens in the server log. This error happens every time the button is clicked.

```
Uncaught ReferenceError: meshcentral2credCallback is not defined
```

![image](https://user-images.githubusercontent.com/60354324/119891809-0a14b800-beff-11eb-8a6e-8955864c458a.png)

[meshcentral-amtconsole-1622146791776.log](https://github.com/Ylianst/MeshCentral/files/6556589/meshcentral-amtconsole-1622146791776.log)

**Comment by Ylianst** [comment]
I have not been able to replicate this, but will try again.

**Comment by ib-mlatin** [comment]
I tried deleting node_modules and re-installed MeshCentral. Still no change. Something must be funky in my config.json file or my MongoDB database. I posted config.json below just in case anyone can see anything that could be causing an issue.

```
{
  "__comment__" : "This is a sample configuration file, edit a section and remove the _ in front of the name. Refer to the user's guide for details.",
  "settings": {
    "Cert": "meshcentral.domain.com",
    "MongoDb": "mongodb://127.0.0.1:27017",
    "_MongoDbName": "meshcentral",
    "_MongoDbChangeStream": true,
    "_WANonly": true,
    "_LANonly": true,
    "_Minify": 1,
    "_SessionTime": 30,
    "SessionKey": "<HIDDEN>",
    "_SessionSameSite": "strict",
    "_DbEncryptKey": "<HIDDEN>",
    "DbRecordsEncryptKey": "<HIDDEN>",
    "_DbRecordsDecryptKey": "<HIDDEN>",
    "_DbExpire": {
      "events": 1728000,
      "powerevents": 864000
    },
    "Port": 8081,
    "RedirPort": 8082,
    "AliasPort": 443,
    "_AllowLoginToken": true,
    "_AllowFraming": true,
    "_WebRTC": false,
    "_Nice404": false,
    "_ClickOnce": false,
    "_SelfUpdate": true,
    "_AgentPing": 60,
    "AgentPong": 300,
    "_AgentIdleTimeout": 150,
    "_MeshErrorLogPath": "c:\\tmp",
    "_NpmPath": "c:\\npm.exe",
    "_NpmProxy": "http://1.2.3.4:80",
    "_AllowHighQualityDesktop": true,
    "_UserAllowedIP": "127.0.0.1,192.168.1.0/24",
    "_UserBlockedIP": "127.0.0.1,::1,192.168.0.100",
    "_AgentAllowedIP": "192.168.0.100/24",
    "_AgentBlockedIP": "127.0.0.1,::1",
    "_LocalDiscovery": {
      "name": "Local server name",
      "info": "Information about this server"
    },
    "TlsOffload": "<HIDDEN>",
    "_MpsPort": 44330,
    "_MpsAliasPort": 4433,
    "_MpsAliasHost": "mps.mydomain.com",
    "MpsTlsOffload": false,
    "_No2FactorAuth": true,
    "_Log": "main,web,webrequest,cert",
    "_WebRtConfig": {
      "iceServers": [
        { "urls": "stun:stun.services.mozilla.com" },
        { "urls": "stun:stun.l.google.com:19302" }
      ]
    },
    "_AutoBackup": {
      "backupIntervalHours": 24,
      "keepLastDaysBackup": 10,
      "zipPassword": "<HIDDEN>",
      "_backupPath": "C:\\backups"
    },
    "_Redirects": {
      "meshcommander": "https://www.meshcommander.com/"
    },
    "__MaxInvalidLogin": "Time in minutes, max amount of bad logins from a source IP in the time before logins are rejected.",
    "MaxInvalidLogin": { "time": 10, "count": 10, "coolofftime": 10 },
    "_Plugins": {
        "enabled": true
    }
  },
  "domains": {
    "": {
      "Title": "domain MeshCentral",
      "Title2": "Device Management",
      "_TitlePicture": "title-sample.png",
      "_UserQuota": 1048576,
      "_MeshQuota": 248576,
      "_NewAccounts": true,
      "_UserNameIsEmail": true,
      "_NewAccountEmailDomains": [ "sample.com" ],
      "_NewAccountsRights": [ "nonewgroups", "notools" ],
      "NewAccounts": false,
      "Footer": "<a href='https://helpdesk.domain.com'>domain HelpDesk</a>",
      "CertUrl": "https://meshcentral.domain.com/",
      "_PasswordRequirements": { "min": 8, "max": 128, "upper": 1, "lower": 1, "numeric": 1, "nonalpha": 1, "reset": 90, "force2factor": true, "skip2factor": "127.0.0.1,192.168.2.0/24" },
      "_AgentNoProxy": true,
      "_GeoLocation": true,
      "_UserAllowedIP": "127.0.0.1,192.168.1.0/24",
      "_UserBlockedIP": "127.0.0.1,::1,192.168.0.100",
      "_AgentAllowedIP": "192.168.0.100/24",
      "_AgentBlockedIP": "127.0.0.1,::1",
      "___UserSessionIdleTimeout__" : "Number of user idle minutes before auto-disconnect",
      "_UserSessionIdleTimeout" : 30,
      "__UserConsentFlags__" : "Set to: 1 for desktop, 2 for terminal, 3 for files, 7 for all",
      "_UserConsentFlags" : 7,
      "_Limits": {
        "_MaxDevices": 100,
        "_MaxUserAccounts": 100,
        "_MaxUserSessions": 100,
        "_MaxAgentSessions": 100,
        "MaxSingleUserSessions": 10
      },
      "auth": "ldap",
      "ldapOptions": {
        "url": "ldaps://corp.domain.com:636",
        "bindDN": "<HIDDEN>",
        "bindCredentials": "<HIDDEN>",
        "searchBase": "CN=Users,DC=corp,DC=domain,DC=com",
        "searchFilter": "(sAMAccountName={{username}})",
        "tlsOptions": {
          "ca": "<HIDDEN>"
        }
      },
      "_AmtAcmActivation": {
        "log": "amtactivation.log",
        "certs": {
          "mycertname": {
            "certfiles": [ "amtacm-leafcert.crt", "amtacm-intermediate1.crt", "amtacm-intermediate2.crt", "amtacm-rootcert.crt" ],
            "keyfile": "amtacm-leafcert.key"
          }
        }
      },
      "_Redirects": {
        "meshcommander": "https://www.meshcommander.com/"
      },
      "_yubikey": { "id": "0000", "secret": "xxxxxxxxxxxxxxxxxxxxx", "_proxy": "http://myproxy.domain.com:80" },
      "_httpheaders": {
        "Strict-Transport-Security": "max-age=360000",
        "x-frame-options": "SAMEORIGIN",
        "Content-Security-Policy": "default-src 'none'; script-src 'self' 'unsafe-inline'; connect-src 'self'; img-src 'self' data:; style-src 'self' 'unsafe-inline'; frame-src 'self'; media-src 'self'"
      },
      "_agentConfig": [ "webSocketMaskOverride=1" ],
      "_SessionRecording": {
        "_filepath": "C:\\temp",
        "_index": true,
        "__protocols__": "Is an array: 1 = Terminal, 2 = Desktop, 5 = Files, 100 = Intel AMT WSMAN, 101 = Intel AMT Redirection",
        "protocols": [ 1, 2, 101 ]
      }
    },
    "_customer1": {
      "_DNS": "customer1.myserver.com",
      "_Title": "Customer1",
      "_Title2": "TestServer",
      "_NewAccounts": 1,
      "_Auth": "sspi",
      "_Footer": "Test",
      "_CertUrl": "https://192.168.2.106:443/"
    },
    "_info": {
      "_share": "C:\\ExtraWebSite"
    }
  },
  "_letsencrypt": {
    "__comment__": "Requires NodeJS 10.12 or better, Go to https://letsdebug.net/ first before trying Let's Encrypt.",
    "email": "myemail@myserver.com",
    "names": "myserver.com,customer1.myserver.com",
    "rsaKeySize": 3072,
    "production": false
  },
  "_peers": {
    "serverId": "server1",
    "servers": {
      "server1": { "url": "wss://192.168.2.133:443/" },
      "server2": { "url": "wss://192.168.1.106:443/" }
    }
  },
  "smtp": {
    "host": "relay.domain.com",
    "port": 25,
    "from": "meshcentral@domain.com",
    "__tls__": "When 'tls' is set to true, TLS is used immidiatly when connecting. For SMTP servers that use TLSSTART, set this to 'false' and TLS will still be used.",
    "tls": false,
    "___tlscertcheck__": "When set to false, the TLS certificate of the SMTP server is not checked.",
    "_tlscertcheck": true,
    "__tlsstrict__": "When set to true, TLS cypher setup is more limited, SSLv2 and SSLv3 are not allowed.",
    "_tlsstrict": true
  }
}
```

**Comment by tonyfairbanks** [comment]
I'm getting the exact same thing. I'm using NGINX as a reverse proxy. Both Firefox and Chrome cause the same errors. 
![image](https://user-images.githubusercontent.com/6955271/120680179-a3901c80-c45f-11eb-868b-67f3a00f5ce0.png)

**Comment by tonyfairbanks** [comment]
config.json:
`{
  "$schema": "http://info.meshcentral.com/downloads/meshcentral-config-schema.json",
  "__comment1__": "This is a simple configuration file, all values and sections that start with underscore (_) are ignored. Edit a section and remove the _ in front of the name. Refer to the user's guide for details.",
  "__comment2__": "See node_modules/meshcentral/sample-config-advanced.json for a more advanced example.",
  "settings": {
    "cert": "meshcentral.domain.com",
    "_LANonly": true,
    "_sessionKey": "MyReallySecretPassword1",
    "port": 4430,
    "aliasPort": 443,
    "redirPort": 8000,
    "TlsOffload": "127.0.0.1",
    "redirAliasPort": 80,
    "__MongoDb": "mongodb://127.0.0.1:27017/meshcentral"
  },
  "domains": {
    "": {
      "certUrl": "https://127.0.0.1:443",
      "_title": "MyServer",
      "_title2": "Servername",
      "_minify": true,
      "_newAccounts": true,
      "_userNameIsEmail": true,
      "showPasswordLogin": false,
      "authStrategies": {
           "saml": {
                 "callbackurl": "https://meshcentral.domain.com/auth-saml-callback",
                 "entityid": "meshcentral",
                 "idpurl": "https://adfs.domain.com/adfs/ls",
                 "cert": "adfs.cer",
                 "disableRequestedAuthnContext": true,
                 "newAccounts": true,
                 "newAccountsUserGroups": [ "ugrp//..removed.." ]
           }
      },
      "siteStyle": 2
    }
  },
  "_letsencrypt": {
    "__comment__": "Requires NodeJS 8.x or better, Go to https://letsdebug.net/ first before trying Let's Encrypt.",
    "email": "myemail@mydomain.com",
    "names": "myserver.mydomain.com",
    "production": false
  }
}`

**Comment by ib-mlatin** [comment]
Alright so after looking at this some more, it seems like it stems from the first two errors preventing the WebUI from completely loading everything it needs. I do not have any Admin Realms defined, and my database was created before these options were present. Most likely somewhere in the upgrade process never set this in the database, which is where we are at now.

**Comment by tonyfairbanks** [comment]
I'm not sure why, but this has started working for me again. All I did was upgrade to the latest version, I didn't change any configuration, database, or reverse proxy settings.

**Comment by ib-mlatin** [comment]
Can confirm as well. This is fixed in 0.8.85.
